content/docs/aws/pipeline.md

```diff
@@ -0,0 +1,196 @@
+    AWS_SECRET_ACCESS_KEY: YOUR_BASE64_SECRET_ACCESS
+  ```
+
+  > Note: To get base64 string, run `echo -n $AWS_ACCESS_KEY_ID | base64`
+
+1. Update deployment `ml-pipeline-ui` to use AWS credential environment viariables by running `kubectl edit deployment ml-pipeline-ui -n kubeflow`.
+
+  ```yaml
+  apiVersion: extensions/v1beta1
+  kind: Deployment
+  metadata:
+    name: ml-pipeline-ui
+    namespace: kubeflow
+    ...
+  spec:
+    template:
+      spec:
+        containers:
+        - env:
+          - name: AWS_ACCESS_KEY_ID
+            valueFrom:
+              secretKeyRef:
+                key: AWS_ACCESS_KEY_ID
+                name: aws-secret
+          - name: AWS_SECRET_ACCESS_KEY
+            valueFrom:
+              secretKeyRef:
+                key: AWS_SECRET_ACCESS_KEY
+                name: aws-secret
+          ....
+          image: gcr.io/ml-pipeline/frontend:0.2.0
+          name: ml-pipeline-ui
+  ```
+
+Here's an example.
+<img src="/docs/images/aws/kfp-viewer-tensorboard.png"
+  alt="Kubeflow Pipeline viewer tensorboard"
+  class="mt-3 mb-3 border border-info rounded">
+
+## Support Tensorboard in Kubeflow Pipeline
+
+ [Tensorboard](/docs/pipelines/sdk/output-viewer/#tensorboard) need some extra settings on AWS.
+
+1. Create a Kubernetes secret `aws-secret` S3 read policies in `kubeflow` namespace, check above example.
+
+1. Create a configmap to store template spec of viewer tensorboard pod. Replace `<your_region>` with your S3 region.
+
+  ```yaml
+  apiVersion: v1
+  kind: ConfigMap
+  metadata:
+    name: ml-pipeline-ui-viewer-template
+  data:
+    viewer-tensorboard-template.json: |
+      {
+          "spec": {
+              "containers": [
+                  {
+                      "env": [
+                          {
+                              "name": "AWS_ACCESS_KEY_ID",
+                              "valueFrom": {
+                                  "secretKeyRef": {
+                                      "name": "aws-secret",
+                                      "key": "AWS_ACCESS_KEY_ID"
+                                  }
+                              }
+                          },
+                          {
+                              "name": "AWS_SECRET_ACCESS_KEY",
+                              "valueFrom": {
+                                  "secretKeyRef": {
+                                      "name": "aws-secret",
+                                      "key": "AWS_SECRET_ACCESS_KEY"
+                                  }
+                              }
+                          },
+                          {
+                              "name": "AWS_REGION",
+                              "value": "<your_region>"
+                          }
+                      ]
+                  }
+              ]
+          }
+      }
+  ```
+
+1. Update deployment `ml-pipeline-ui` to use this sepc by running `kubectl edit deployment ml-pipeline-ui -n kubeflow`.
```

Comment:
Update the `ml-pipeline-ui` deployment to use the ConfigMap by running `kubectl edit deployment ml-pipeline-ui -n kubeflow`.

Is there anything else that people should know about running this command? How is the following YAML file related?

Reply:
The following Yaml indicates the changes user need to make after running `edit`